pkg/cmd/job/job_add.go

```diff
@@ -0,0 +1,72 @@
+package job
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/dcos/dcos-cli/api"
+	"github.com/dcos/dcos-core-cli/pkg/metronome"
+	"github.com/spf13/cobra"
+	"golang.org/x/crypto/ssh/terminal"
+)
+
+// newCmdClusterAdd creates a new job.
+func newCmdJobAdd(ctx api.Context) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "add",
+		Short: "add a job",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			client, err := metronomeClient(ctx)
+			if err != nil {
+				return err
+			}
+
+			// Handling input from file or stdin
+			var reader io.Reader
+			switch len(args) {
+			case 0:
+				if terminal.IsTerminal(int(os.Stdin.Fd())) {
```

Comment:
It should rely on `ctx.Input()` (would need a conversion from io.Reader to os.File).